components/datadog/fakeintake/ecsFargate.go

```diff
@@ -14,41 +16,101 @@ const (
 	port          = 80
 )
 
-const (
-	DefaultFakeintakePort = 80
-)
+type Instance struct {
+	pulumi.ResourceState
 
-func FargateLinuxTaskDefinition(e aws.Environment, name string) (*ecs.FargateTaskDefinition, error) {
-	return ecs.NewFargateTaskDefinition(e.Ctx, e.Namer.ResourceName(name), &ecs.FargateTaskDefinitionArgs{
-		Containers: map[string]ecs.TaskDefinitionContainerDefinitionArgs{
-			containerName: *fargateLinuxContainerDefinition(),
-		},
-		Cpu:    pulumi.StringPtr("256"),
-		Memory: pulumi.StringPtr("512"),
-		ExecutionRole: &awsx.DefaultRoleWithPolicyArgs{
-			RoleArn: pulumi.StringPtr(e.ECSTaskExecutionRole()),
+	Host pulumi.StringOutput
+}
+
+func NewECSFargateInstance(e aws.Environment) (*Instance, error) {
+	namer := e.Namer.WithPrefix("fakeintake")
+	opts := []pulumi.ResourceOption{e.WithProviders(config.ProviderAWS, config.ProviderAWSX)}
+
+	instance := &Instance{}
+	if err := e.Ctx.RegisterComponentResource("dd:fakeintake", namer.ResourceName("grp"), instance, opts...); err != nil {
+		return nil, err
+	}
+
+	opts = append(opts, pulumi.Parent(instance))
+
+	alb, err := lb.NewApplicationLoadBalancer(e.Ctx, namer.ResourceName("lb"), &lb.ApplicationLoadBalancerArgs{
+		Name:           e.CommonNamer.DisplayName(pulumi.String("fakeintake")),
+		SubnetIds:      pulumi.ToStringArray(e.DefaultSubnets()),
+		Internal:       pulumi.BoolPtr(!e.ECSServicePublicIP()),
+		SecurityGroups: pulumi.ToStringArray(e.DefaultSecurityGroups()),
+		DefaultTargetGroup: &lb.TargetGroupArgs{
+			Name:       e.CommonNamer.DisplayName(pulumi.String("fakeintake")),
+			Port:       pulumi.IntPtr(port),
+			Protocol:   pulumi.StringPtr("HTTP"),
+			TargetType: pulumi.StringPtr("ip"),
+			VpcId:      pulumi.StringPtr(e.DefaultVPCID()),
 		},
-		TaskRole: &awsx.DefaultRoleWithPolicyArgs{
-			RoleArn: pulumi.StringPtr(e.ECSTaskRole()),
+		Listener: &lb.ListenerArgs{
+			Port:     pulumi.IntPtr(port),
+			Protocol: pulumi.StringPtr("HTTP"),
 		},
-		Family: e.CommonNamer.DisplayName(pulumi.String("fakeintake-ecs")),
-	}, e.WithProviders(config.ProviderAWS, config.ProviderAWSX))
-}
+	}, opts...)
```

Comment:
This code is very close to https://github.com/DataDog/test-infra-definitions/pull/225/files#diff-5368ecf6a73f221e6f5376fbae6989ec734652bba2afbf1e3428aca810fe796cR29-R43. What do you think about having a helper function to reduce the verbosity of the code?

Reply:
It’s very close because it’s the creation of the same resource type.
And it’s already a single function call (`lb.NewLoadBalancer`), but most arguments change.
![image](https://github.com/DataDog/test-infra-definitions/assets/1437785/3ac568fa-d8cc-4a20-870d-d77ecf0269b6)
I’m not sure that replacing one pulumi standard function call by our own custom function with as many arguments, but in flat instead of being structured would provide a big gain.